stats/pp_stage_goal.php

```diff
@@ -0,0 +1,81 @@
+<?php
+$relPath = "./../pinc/";
+include_once($relPath.'base.inc');
+include_once($relPath.'stages.inc');
+include_once($relPath.'graph_data.inc');
+include_once($relPath.'slim_header.inc');
+include_once($relPath.'page_tally.inc');
+
+// Get start of current month as unixtime
+$m_start = mktime(0, 0, 0, date('m'), 1, date('Y'));
+
+// We want the "PP pages goal" to be equal to the current month's F2 actuals
+$page_offset = 0;
+$site_stats = get_site_page_tally_summary("F2");
+$pp_page_goal = $site_stats->curr_month_actual + $page_offset;
+
+// Start with creating the Graph, this enables the use of the cache
+// where possible
+$width = 120;
+$height = 200;
+
+// Get the total pages for projects that have posted
+$page_res = DPDatabase::query(sprintf("
+    SELECT 
+    SUM(n_pages)
+    FROM projects
+    WHERE state='proj_submit_pgposted'
+    AND modifieddate >= %d
+", $m_start));
+
+$row = mysqli_fetch_row($page_res);
+$pp_pages_total = $row[0];
+
+// calculate the goal percent as long as $pp_page_goal isn't zero
+if ($pp_page_goal != 0) {
+    $goal_percent = ceil(($pp_pages_total / $pp_page_goal) * 100);
+} else {
+    $goal_percent = 0;
+}
+
+// Some graph variables
+$datax = [_("F2"), _("PP")];
```

Comment:
This is why the graph was in noncvs -- other sites in theory might not have a F2 if they've changed the stages to only have 1 (or 3!) formatting arounds. We can mitigate this somewhat by ensuring that we are comparing against the last formatting round:
```php
# not exactly elegant, but quick and clear:
$last_formatting_round = Null;
foreach($Round_for_round_id_ as $id => $round) {
    if(is_formatting_round($round)) {
        $last_formatting_round = $id;
    }
}
```

Also: F2 and PP should not be translated as they are effectively language-agnostic keys.

Reply:
Does it have to be the last formatting round, or just the last **_round_** before the project goes into PP?

Reply:
Probably the latter, good point. So perhaps:
```php
$round_before_PP = Null;
foreach($Round_for_round_id_ as $id => $round) {
    if($round->id == "PP") {
        break;
    }
    $round_before_PP = $round->id;
}
```